crates/superposition/src/webhooks/handlers.rs

```diff
@@ -0,0 +1,196 @@
+            }
+            _ => {
+                log::error!("Webhook creation failed with error: {e}");
+                Err(unexpected_error!(
+                    "An error occured please contact the admin."
+                ))
+            }
+        },
+    }
+}
+
+#[get("")]
+async fn list_webhooks(
+    db_conn: DbConnection,
+    filters: Query<PaginationParams>,
+) -> superposition::Result<Json<PaginatedResponse<Webhook>>> {
+    let DbConnection(mut conn) = db_conn;
+
+    let (total_pages, total_items, data) = match filters.all {
+        Some(true) => {
+            let result: Vec<Webhook> = dsl::webhook.get_results(&mut conn)?;
+            (1, result.len() as i64, result)
+        }
+        _ => {
+            let n_functions: i64 = dsl::webhook.count().get_result(&mut conn)?;
+            let limit = filters.count.unwrap_or(10);
+            let mut builder = dsl::webhook
+                .into_boxed()
+                .order(webhook::last_modified_at.desc())
+                .limit(limit);
+            if let Some(page) = filters.page {
+                let offset = (page - 1) * limit;
+                builder = builder.offset(offset);
+            }
+            let result: Vec<Webhook> = builder.load(&mut conn)?;
+            let total_pages = (n_functions as f64 / limit as f64).ceil() as i64;
+            (total_pages, n_functions, result)
+        }
+    };
+
+    Ok(Json(PaginatedResponse {
+        total_pages,
+        total_items,
+        data,
+    }))
+}
+
+#[get("/{webhook_name}")]
+async fn get(
+    params: web::Path<WebhookName>,
+    db_conn: DbConnection,
+) -> superposition::Result<Json<Webhook>> {
+    let DbConnection(mut conn) = db_conn;
+    let w_name: String = params.into_inner().into();
+    let webhooks = fetch_webhook(&w_name, &mut conn)?;
+    Ok(Json(webhooks))
+}
+
+#[put("/{webhook_name}")]
+async fn update(
+    params: web::Path<WebhookName>,
+    db_conn: DbConnection,
+    user: User,
+    request: web::Json<CreateWebhookRequest>,
+) -> superposition::Result<Json<Webhook>> {
+    let DbConnection(mut conn) = db_conn;
+    let req = request.into_inner();
+    let w_name: String = params.into_inner().into();
+    let events: Vec<String> = req
+        .events
+        .into_iter()
+        .map(|event| event.to_string())
+        .collect();
+
+    validate_events(&events, Some(&w_name), &mut conn)?;
+
+    let update = diesel::update(dsl::webhook)
+        .filter(webhook::name.eq(w_name))
+        .set((
+            webhook::description.eq(req.description),
+            webhook::enabled.eq(req.enabled),
+            webhook::url.eq(req.url),
+            webhook::method.eq(req.method),
+            webhook::version.eq(req.version.unwrap_or("v1".to_owned())),
+            webhook::custom_headers.eq(req.custom_headers),
+            webhook::events.eq(events),
+            webhook::last_modified_by.eq(user.email),
+            webhook::last_modified_at.eq(Utc::now()),
+        ))
```

Comment:
instead of manually setting all the values from request, simply derive `AsChangeset` on the `UpdateWebhookRequest` type which you are going to add, and so:
```suggestion
        .set((
            req,
            webhook::last_modified_by.eq(user.email),
            webhook::last_modified_at.eq(Utc::now()),
        ))
```